core/server/common/src/main/java/alluxio/util/FeatureUtils.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.util;
+
+import alluxio.conf.PropertyKey;
+import alluxio.conf.ServerConfiguration;
+import alluxio.master.journal.JournalType;
+
+/**
+ * Utilities to detect features that Alluxio is running with.
+ */
+public final class FeatureUtils {
+  /**
+   * Utility to check the master journal type is embedded journal.
+   *
+   * @return true, if running with embedded journal
+   */
+  public static boolean isEmbeddedJournal() {
+    return JournalType.EMBEDDED.toString()
+        .equals(ServerConfiguration.get(PropertyKey.MASTER_JOURNAL_TYPE));
+  }
+
+  /**
+   * Utility to check the master metastore is rocks.
+   *
+   * @return true, if running with rocks
+   */
+  public static boolean isRocks() {
+    return "ROCKS".equals(ServerConfiguration.get(PropertyKey.MASTER_METASTORE));
+  }
+
+  /**
+   * Utility to check Zookeeper is enabled.
+   *
+   * @return true, if Zookeeper is enabled
+   */
+  public static boolean isZookeeperEnable() {
```

Comment:
instead of having a method for each property, can we have a single method for boolean properties that takes in the argument for the property key? this would make it easier to reuse in the future

Reply:
sorry, i dont understand the reasoning behind why we shouldn't be more efficient in writing methods in utils class?
what is the issue with having:
```
public static boolean isBooleanPropertyEnabled(String propertyKey) {
  return ServerConfiguration.getBoolean(propertyKey);
}
```
to replace all of the simple boolean check methods?

Reply:
I am not sure I understand the difference between using the existing method compare to change to `isBooleanPropertyEnabled`. If we want to abstract a method like `isBooleanPropertyEnabled`, we can directly use `ServerConfiguration.getBoolean(propertyKey)` instead.

Reply:
honestly that sounds like a better idea than mine